packages/quantic/force-app/main/default/lwc/quanticUserActionsToggle/__tests__/quanticUserActionsToggle.test.js

```diff
@@ -0,0 +1,211 @@
+    if (element instanceof QuanticUserActionsToggle) {
+      element.setInitializationError();
+    }
+  };
+}
+
+function cleanup() {
+  // The jsdom instance is shared across test cases in a single file so reset the DOM
+  while (document.body.firstChild) {
+    document.body.removeChild(document.body.firstChild);
+  }
+  jest.clearAllMocks();
+  isInitialized = false;
+}
+
+describe('c-quantic-user-actions-toggle', () => {
+  beforeAll(() => {
+    mockSuccessfulHeadlessInitialization();
+  });
+
+  afterEach(() => {
+    cleanup();
+  });
+
+  it('should display the user actions toggle button', async () => {
+    const element = createTestComponent();
+    await flushPromises();
+
+    const userActionsToggleButton = element.shadowRoot.querySelector(
+      selectors.userActionsToggleButton
+    );
+    const userActionsToggleButtonIcon = element.shadowRoot.querySelector(
+      selectors.userActionsToggleButtonIcon
+    );
+
+    expect(userActionsToggleButton).not.toBeNull();
+    expect(userActionsToggleButton.title).toBe(userActionsLabel);
+    expect(userActionsToggleButtonIcon.iconName).toBe(userActionsIcon);
+  });
+
+  it('should log analytics when the user action modal is opened', async () => {
+    const element = createTestComponent();
+    await flushPromises();
+
+    const userActionsToggleButton = element.shadowRoot.querySelector(
+      selectors.userActionsToggleButton
+    );
+
+    expect(userActionsToggleButton).not.toBeNull();
+    await userActionsToggleButton.click();
+    await flushPromises();
+
+    expect(functionsMocks.logOpenUserActions).toHaveBeenCalledTimes(1);
+  });
+
+  it('should properly display the user actions modal header', async () => {
+    const element = createTestComponent();
+    await flushPromises();
+
+    const modalHeaderIcon = element.shadowRoot.querySelector(
+      selectors.modalHeaderIcon
+    );
+    const modalHeaderLabel = element.shadowRoot.querySelector(
+      selectors.modalHeaderLabel
+    );
+
+    expect(modalHeaderIcon).not.toBeNull();
+    expect(modalHeaderLabel).not.toBeNull();
+
+    expect(modalHeaderIcon.iconName).toBe(userActionsIcon);
+    expect(modalHeaderIcon.title).toBe(userActionsLabel);
+    expect(modalHeaderLabel.textContent).toBe(userActionsLabel);
+  });
+
+  it('should properly display the user actions timeline inside the modal', async () => {
+    const element = createTestComponent();
+    await flushPromises();
+
+    const userActionsTimeline = element.shadowRoot.querySelector(
+      selectors.userActionsTimeline
+    );
+
+    expect(userActionsTimeline).not.toBeNull();
+    expect(userActionsTimeline.engineId).toBe(exampleEngineId);
+    expect(userActionsTimeline.userId).toBe(exampleUserId);
+    expect(userActionsTimeline.ticketCreationDateTime).toBe(
+      exampleTicketCreationDateTime
+    );
+    expect(userActionsTimeline.excludedCustomActions).toEqual(
+      exampleExcludedCustomActions
+    );
+  });
```

Comment:
maybe this can be put into a describe block "when clicking on the user actions toggle button" or "when opening the user actions modal"